backend/src/interactions/updateUnconfirmedContribution/AbstractUnconfirmedContribution.role.ts

```diff
@@ -45,14 +57,31 @@ export abstract class AbstractUnconfirmedContributionRole {
   // third, actually update entity
   protected abstract update(): void
 
+  protected wasUpdateAlreadyCalled(): boolean {
+    return this.currentStep > 3
+  }
+
   // call all steps in order
   public async checkAndUpdate(context: Context): Promise<void> {
     if (!context.user || !context.role) {
       throw new LogError('missing user or role on context')
     }
+    this.currentStep = 1
     this.checkAuthorization(context.user, context.role)
+    this.currentStep = 2
     await this.validate(getClientTimezoneOffset(context))
+    this.currentStep = 3
     this.update()
+    this.currentStep = 4
+  }
+
+  public createContributionMessage(): ContributionMessageBuilder | undefined {
+    // must be called before call at update
+    if (this.wasUpdateAlreadyCalled()) {
+      throw new LogError('please call before call of checkAndUpdate')
```

Comment:
I don't believe that this errormessage is expressiv for an user to know what he have to do, right?

Reply:
Yes but this is not for the user, it is for the developer using this role.
Should I change it to Logical Error for user and write the details into log? But on unit-tests logs not shown so it isn't clear was is going wrong.
The user cannot do anything here. Maybe I find a better approach to make using this class fault proof in future.